ruleguard/gorule.go

```diff
@@ -67,3 +71,37 @@ func (params *filterParams) typeofNode(n ast.Node) types.Type {
 
 	return types.Typ[types.Invalid]
 }
+
+func newState() *State {
+	env := quasigo.NewEnv()
+	state := &State{
+		env:      env,
+		importer: newGoImporter(),
+		typeByFQN: map[string]types.Type{
+			// Predeclared types.
+			`error`:      types.Universe.Lookup("error").Type(),
+			`bool`:       types.Typ[types.Bool],
+			`int`:        types.Typ[types.Int],
+			`int8`:       types.Typ[types.Int8],
+			`int16`:      types.Typ[types.Int16],
+			`int32`:      types.Typ[types.Int32],
+			`int64`:      types.Typ[types.Int64],
+			`uint`:       types.Typ[types.Uint],
+			`uint8`:      types.Typ[types.Uint8],
+			`uint16`:     types.Typ[types.Uint16],
+			`uint32`:     types.Typ[types.Uint32],
+			`uint64`:     types.Typ[types.Uint64],
+			`uintptr`:    types.Typ[types.Uintptr],
+			`string`:     types.Typ[types.String],
+			`float32`:    types.Typ[types.Float32],
+			`float64`:    types.Typ[types.Float64],
+			`complex64`:  types.Typ[types.Complex64],
+			`complex128`: types.Typ[types.Complex128],
```

Comment:
do we need all this types? or we can limit to `int64` and/or `float64` ?

Reply:
These types are needed so you can write:
```go
ctx.GetType(`float32`)
types.Typ[types.Float32]
```
In other words, it's needed for the filter implementation on the user side.

It's not 100% necessary as we could just export the same `types.Typ` map and all type kind constants, but we'll still need `GetType` for more complex cases:
```go
// may dynamically load "fmt" package, finds `Stringer` type in it, caches the result
ctx.GetType(`fmt.Stringer`)
```
It also works for 3rd party packages with FQN:
```go
// may dynamically load "github.com/foo/bar" package, finds `Baz` type in it, caches the result
ctx.GetType(`github.com/foo/bar.Baz`)
```